lib/python/ensembl/brc4/runnable/manifest_stats.py

```diff
@@ -143,25 +143,40 @@ def parse_gff3(self, gff3_handle: TextIO) -> List:
         biotypes = {}
 
         for rec in GFF.parse(gff3_handle):
-            for feat in rec.features:
-                self.increment_biotype(biotypes, feat)
-                for feat2 in feat.sub_features:
-                    self.increment_biotype(biotypes, feat2)
+            for feat1 in rec.features:
+                is_protein = False
+                for feat2 in feat1.sub_features:
+                    if feat2.type == "mRNA":
+                        is_protein = True
+                    self.increment_biotype(biotypes, feat2.id,  f"{feat1.type}-{feat2.type}")
                     for feat3 in feat2.sub_features:
-                        self.increment_biotype(biotypes, feat3)
+                        if feat3.type == 'exon':
+                            continue
+                        self.increment_biotype(biotypes, feat3.id, f"{feat1.type}-{feat2.type}-{feat3.type}")
+                if is_protein:
+                    self.increment_biotype(biotypes, feat1.id, f"PROT_{feat1.type}")
+                else:
+                    self.increment_biotype(biotypes, feat1.id, f"NONPROT_{feat1.type}")
+                
+                if feat1.type in ("gene", "pseudogene"):
+                    self.increment_biotype(biotypes, feat1.id, "ALL_GENES")
                 
         # Order
         sorted_biotypes = OrderedDict()
         for name in sorted(biotypes.keys()):
-            sorted_biotypes[name] = biotypes[name]
+            data = biotypes[name]
+            counter = Counter(data["ids"])
+            data["unique_count"] = len(counter.keys())
+            sorted_biotypes[name] = data
         
-        stats = ["%9d\t%20s\tID = %s" % (data["count"], biotype, data["example"]) for (biotype, data) in sorted_biotypes.items()]
+        stats = [f"{data['unique_count']:>9}\t{biotype:<20}\tID = {data['example']}" for (biotype, data) in sorted_biotypes.items()]
         
         return stats
 
-    def increment_biotype(self, biotypes: Dict, feature) -> None:
-        biotype = feature.type
-        if not biotype in biotypes:
-            biotypes[biotype] = { "count" : 0, "example" : feature.id }
-        biotypes[biotype]["count"] += 1
+    def increment_biotype(self, biotypes: Dict, feature_id: str, feature_biotype: str) -> None:
```

Comment:
This method seems to be related to this class, but does not need to know anything about it:
```suggestion
    @staticmethod
    def increment_biotype(biotypes: Dict, feature_id: str, feature_biotype: str) -> None:
```
If you ahead with this change, all the previous call should be replaced from `self.increment_biotype()` to `manifest_stats.increment_biotype()`.

Reply:
Done